text/0010-bft-signatures.md

```diff
@@ -0,0 +1,291 @@
+2. Else, the orderer that blocks are pulled from is not withholding newly formed blocks. However, it is possible that new blocks are not formed due to lack of client activity. It is imperative to ensure that malicious ordering nodes cannot make honest ordering nodes be falsely suspected of block withholding.
+As each "block existence attestation" contains `2f+1` signatures it is impossible to forge one. Hence, if a node sends a attestation for block existence, it means this a block with such sequence number was indeed appended to the blockchain.
+
+In order for a peer or an orderer to convey the intent of receiving only blocks without transactions, 
+the [SeekInfo](https://github.com/hyperledger/fabric-protos/blob/main/orderer/ab.proto#L45) message is expanded to contain also a `SeekContentType` field:
+
+```
+  // SeekContentType indicates what type of content to deliver in response to a request. If BLOCK is specified,
+  // the orderer will stream blocks back to the peer. This is the default behavior. If HEADER is  specified, the
+  // orderer will stream only a the header and the signature, and the payload field will be set to nil. This allows
+  // the requester to ascertain that the respective signed block exists in the orderer (or cluster of orderers).
+  enum SeekContentType {
+      BLOCK = 0;
+      HEADER_WITH_SIG =1;
+   }
+```
+
+By making the default value be `BLOCK` the API is backward compatible to old consumers.
+
+Indeed, the SmartBFT's approach for Byzantine fault tolerant block replication overcomes the linear overhead of the naive approach. However, a malicious ordering node can still withhold a block up until right before the suspicion timeout expires at the party that pulls the block.
+Hence, while a malicious orderer may not withhold blocks indefinitely, it may slow down the replication of its consumers.
+
+Another minor inconvenience is that the peer admin needs to configure the peer correctly to use the Byzantine fault tolerant implementation instead of the regular one.
+
+### Signature identity referencing
+
+In SmartBFT, like Raft, each configuration block contains a set of consenter (ordering service consensus protocol participant) definitions. 
+Each such consenter definition contains various information that is used either to connect or to authenticate the consenter:
+
+```
+message Consenter {
+    uint64 consenter_id = 1;
+    string host = 2;
+    uint32 port = 3;
+    string msp_id = 4;
+    bytes identity = 5;
+    bytes client_tls_cert = 6;
+    bytes server_tls_cert = 7;
+}
+```
+
+When validating a signature signed by orderer with numerical identifier **j**, the mapping between **j** and the matching identity (which carries the public key) is taken from the latest configuration block.
+
+Recall, the naive approach to encode several signatures on a block in its metadata involves encoding an amount of orderer identities that is linear in the number of ordering nodes.
```

Comment:
if I am correct, we avoid linear space in block bandwidth by `Consenter` structure, and we can't avoid linear space stored in peer/orderer, as we use x509 cert to verify msg(in BFT), as peer/orderer need translate from `Consenter` to find a specific x509 cert in memory/storage for verification works.